### PR TITLE
PE-356 Rename team channels

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description
- rocket.py  line 631: Change the way how the team names are created [jira ticket](https://proversity.atlassian.net/browse/PE-356)
- rocketc.js line 148-149 remove the character "/" from the name because the url is created in the following way `http://192.168.0.36:3000/group/SecondTeam11(Eng%252Fteam11)__coursev1edxcs1082019?layout=embedded` for the name `SecondTeam11(Eng/team11)__coursev1edxcs1082019`
- Other changes: query parameter support was removed so this changes allow to filter by team, topic and course.

### CODE REVIEW

File: rocketc/static/js/src/rocketc.js [hunk 1]
```diff
@@ -145,7 +145,8 @@ function RocketChatXBlock(runtime, element) {
             $("#select-channel").val($("#tool-buttons").attr("data-default"));
         }
         $("#select-channel").change(function(){
-            url = $("#tool-buttons").attr("data-domain") + $("#select-channel").val() +"?layout=embedded";
+            let groupName = $("#select-channel").val().replace("/", "%252F")
```
Comment (Squirrel18) [inline rocketc/static/js/src/rocketc.js]: let variables only works in browers with es6 support. @andrey-canon

File: rocketc/static/js/src/rocketc.js [hunk 2]
```diff
@@ -145,7 +145,8 @@ function RocketChatXBlock(runtime, element) {
...
+            url = $("#tool-buttons").attr("data-domain") + groupName +"?layout=embedded";
```
Comment (Squirrel18) [inline rocketc/static/js/src/rocketc.js]: Missing space after last plus sign. @andrey-canon

File: rocketc/rocketc.py [hunk 3]
```diff
@@ -718,10 +714,15 @@ def _get_list_groups(self, user_id, auth_token, **kwargs):
...
         groups = api.list_all_groups(user_id, auth_token, **kwargs)
+        team, topic = generate_team_variables(self._get_team(self.user_data["username"]))
         if groups["success"]:
-            groups = groups["groups"]
-            for group in groups:
-                yield group["name"]
+            for group in groups["groups"]:
+                fields = group.get("customFields", {})
+                try:
+                    if (team == fields["team"] and topic == fields["topic"] and self.course_id == fields["course"]):
```
Comment (Squirrel18) [inline rocketc/rocketc.py]: Could you use .get to obtain the values from the 'fields' dict. @andrey-canon

File: rocketc/rocketc.py [hunk 4]
```diff
@@ -718,10 +714,15 @@ def _get_list_groups(self, user_id, auth_token, **kwargs):
...
+                            yield group["name"]
+                except IndexError:
```
Comment (Squirrel18) [inline rocketc/rocketc.py]: If you want to catch a missing key from a dict, the correct error is a KeyError not an IndexError. @andrey-canon

File: rocketc/rocketc.py [hunk 5]
```diff
@@ -628,7 +628,7 @@ def create_group(self, data, suffix=""):
     def _create_team_group_name(team, group_name, course):
 
         team_name, team_topic = generate_team_variables(team)
-        return "{}__{}__{}__{}".format(group_name, team_name, team_topic, course)
+        return "{}({}/{})__{}".format(group_name, team_topic, team_name, course)
```
Comment (Squirrel18) [inline rocketc/rocketc.py]: Could you please add an example comment of the formating that you are doing here? @andrey-canon

File: rocketc/static/js/src/rocketc.js (hunk 1 above)
Comment (andrey-canon) [reply rocketc/static/js/src/rocketc.js]: changed to var

File: rocketc/static/js/src/rocketc.js (hunk 2 above)
Comment (andrey-canon) [reply rocketc/static/js/src/rocketc.js]: added

File: rocketc/rocketc.py (hunk 4 above)
Comment (andrey-canon) [reply rocketc/rocketc.py]: changed

File: rocketc/rocketc.py (hunk 5 above)
Comment (andrey-canon) [reply rocketc/rocketc.py]: Please read the jira ticket added in the description

Comment (Squirrel18) [reply rocketc/rocketc.py]: It's not for me, it's for who will read the code in the future. @andrey-canon

Comment (Squirrel18) [reply rocketc/rocketc.py]: It's not for me, it's for who will read the code in the future. @andrey-canon

**Comment by Squirrel18** [comment]
Just a single commnet. @andrey-canon

**Comment by andrey-canon** [comment]
@Squirrel18  Done

File: rocketc/rocketc.py (hunk 5 above)
Comment (andrey-canon) [reply rocketc/rocketc.py]: done

**Comment by Squirrel18** [comment]
Please @andrey-canon Squash and merge.

**Pull request merged**